components/shared/TextInput.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { StyleSheet, View, TextInput } from 'react-native';
+import Colors from '../../constants/Colors';
+import { Text } from './Typography';
+
+export default function Input(props) {
+  const { label, touched, errors, ...inputProps } = props;
+
+  return (
+    <View style={styles.container}>
+      {label && (
+        <Text S bold style={styles.label}>
+          {label}
+        </Text>
+      )}
+      <TextInput
+        autoCorrect={false}
+        keyboardAppearance="dark"
+        returnKeyType="done"
+        placeholderTextColor="rgba(255, 255, 255, 0.5)"
+        style={styles.input}
+        {...inputProps}
+      />
+      {errors && touched && (
```

Comment:
if the input has an error and it's been visited, then we'll show the error message from our validation function